src/pywifes/wifes_calib.py

```diff
@@ -960,54 +961,77 @@ def derive_wifes_telluric(
-        sp2.set_title("Individual H2O Telluric Correction functions")
-        sp2.legend(loc="lower left", fancybox=True, shadow=True)
-        sp2.set_xlim([numpy.min(base_wave), numpy.max(base_wave)])
-        pylab.figure()
-        pylab.plot(base_wave, final_O2_corr, color="r", lw=2, label="Default O2 lines")
-        pylab.plot(
-            base_wave, final_H2O_corr, color="k", lw=2, label="Default H2O lines"
+    # Check Plot
+    if plot:
+
+        fig, (ax_top, ax_bottom) = plt.subplots(2, 1, sharex=True, figsize=(10, 8))
+        plt.suptitle("Telluric Correction Function", size=16)
+
+        # Top subplot (Telluric start)
+        ax_top.plot(obs_wave, obs_flux, "C0", label="Observed Flux")
+        ax_top.plot(obs_wave, smooth_cont, "r", ls="dashed", label="Smooth Fitting")
+        ax_top.legend()
+        ax_top.set_ylabel(r"Flux [$F_{\lambda}$]")
+
+        # Set y-limits to exclude peaks
+        lower_limit = numpy.percentile(obs_flux, 0.2)
+        upper_limit = numpy.percentile(obs_flux, 99.8)
+        ax_top.set_ylim(lower_limit, upper_limit)
+
+        # Bottom subplot (Telluric correction derived)
+        # Mask the continum for plotting
+        telluric_correction = final_O2_corr * final_H2O_corr
+
+        ax_bottom.plot(
+            base_wave,
+            telluric_correction,
+            color="k",
+            lw=1,
+            label=r"Telluric correction",
         )
-        pylab.xlabel(r"Wavelength [$\AA$]")
-        pylab.legend(loc="lower left", fancybox=True, shadow=True)
-        pylab.xlim([numpy.min(base_wave), numpy.max(base_wave)])
-        pylab.title("Telluric Correction functions")
-        if savefigs:
-            save_fn = save_prefix + "final_solutions.png"
-            pylab.savefig(save_fn)
-            save_fn_1 = save_prefix + "O2_corrections.png"
-            fig1.savefig(save_fn_1)
-            save_fn_2 = save_prefix + "H2O_corrections.png"
-            fig2.savefig(save_fn_2)
-        if plot:
-            pylab.show()
+        # Shading the masked regions for O2 and H2O corrections
+        ax_bottom.fill_between(
+            base_wave,
+            final_O2_corr,
+            where=(final_O2_corr < 1),
+            lw=0,
+            color="C4",
+            alpha=0.5,
+            label=r"O$_2$ lines",
+        )
+        ax_bottom.fill_between(
+            base_wave,
+            final_H2O_corr,
+            where=(final_H2O_corr < 1),
+            lw=0,
+            color="C0",
+            alpha=0.5,
+            label=r"H$_2$O lines",
+        )
+
+        # for i, cube_fn in enumerate(cube_fn_list):
+        #     airmass = airmass_list[i]
+        #     wave, O2_ratio = O2_corrections[i]
+        #     H2O_ratio = H2O_corrections[i][1]
+
+        #     ax_bottom.plot(wave, O2_ratio**(1.0/(airmass**O2_power)), label=f'{cube_fn} O2')
+        #     ax_bottom.plot(wave, H2O_ratio**(1.0/(airmass**H2O_power)), label=f'{cube_fn} H2O')
```

Comment:
Is it useful somehow all this commented out code? If not, remove it, please.

Reply:
Yeah, this is coming from the refactorization of the plots logic. I'll leave it as it is for now.